XCUITests/FxScreenGraph.swift

```diff
@@ -115,6 +115,7 @@ class Action {
     static let ReloadURL = "ReloadURL"
 
     static let TogglePrivateMode = "TogglePrivateBrowing"
+    static let TogglePrivateModeDirectAccess = "TogglePrivateModeDirectAccess"
```

Comment:
I'd add a third action: `TogglePrivateModeFromTabTray`.

In the screenStates:

```swift
// In the tab
screenState.tap(app.buttons["Private Mode"], forAction: Action.TogglePrivateModeFromTabBar, Action.TogglePrivateMode, if: "tablet == true")

// in the tab tray (with the relevant accessibilityIdentifier)
screenState.tap(app.buttons["Private Mode"], forAction: Action.TogglePrivateModeFromTabTray, Action.TogglePrivateMode)
```

Then you can put the common logic to change the user state from an action on the map:
 
```swift
map.addScreenAction(Action.TogglePrivateBrowsing) { userState in 
     userState.isPrivate = !userState.isPrivate
}
```

Now the `navigator` will chose the fasted route on the iPad.

Reply:
Hi @jhugman thanks for your comments! I'm trying to apply them but there are some errors...
>// In the tab
screenState.tap(app.buttons["Private Mode"], forAction: Action.TogglePrivateModeFromTabBar, Action.TogglePrivateMode, if: "tablet == true")

// in the tab tray (with the relevant accessibilityIdentifier)
screenState.tap(app.buttons["Private Mode"], forAction: Action.TogglePrivateModeFromTabTray, Action.TogglePrivateMode) >

I can't use the same action in different screenStates, for example in TabTray or NewTabScreen, there is an error about the action defined before and that has to be unique:
>Action TogglePrivateModeTabBar is defined elsewhere, but should be unique>

Also, in order to use the common logic to change the user state, I get an error: 
>Ambiguous reference to member 'addScreenAction(_:transitionTo:file:line:recorder:)'<

Looks like I need the transitionTo, but in this case, I just want to change the userState don't need to change the current screen

Reply:
In order to get a fix I thought about having a common func 
```func makePrivateModeAvailableTabBar() {
screenState.tap(app.buttons["Private Mode"], forAction: Action.TogglePrivateModeTabBar, if: "tablet == true") { userState in
            userState.isPrivate = !userState.isPrivate
        }}```

Would that be a correct solution? 
Thanks @jhugman!